### ISSUE TITLE
Life After Dependency Injection

### ISSUE THREAD

**Issue by mayoatte** [Azure/azure-functions-host#4411]
#### Question
I'm trying to get a handle on scenarios that are unblocked / enabled by the release of Azure Functions DI (#3736). Creating a new issue as that one is closed (and possibly over subscribed). 

Anyways, i'm interested in:

- Existing issues that can now be fixed because this feature is available
- New feature requests that are now unblocked
- Interesting uses of this feature from the community

Some related issues / features/ capabilities i remember are:

* Function Filters
  * Can they be improved / made non obsolete now?
* Telemetry Processor (#3741)
* Adding context to request telemetry?
  * Needed access to built in telemetry client i believe
* Distributed tracing in App Insights
  * Needed access to request telemetry

Thanks, all!

**Comment by pragnagopa** [comment]
@brettsam / @jeffhollan / @fabiocav  - Do we have documentation to answer above questions?

**Comment by mayoatte** [comment]
@fabiocav 

Here's an example scenario i ran into recently that i thought would be possible after the DI release.

I was trying to inject a scoped service (IExecutionContext) into a function filter (a singleton) via the delegate factory method (see screenshot below). But the IExecutionContext instance that is resolved in the function filter is different from the one resolved elsewhere (during the same request). The overall goal here is to set this execution context in the filter and use elsewhere.

![image](https://user-images.githubusercontent.com/29760620/58363572-51bd6080-7e5b-11e9-881a-6bbca07d82a6.png)

Is this an invalid scenario for DI in Azure Functions?

**Comment by meixger** [comment]
Related? https://github.com/Azure/azure-functions-host/issues/3399

**Comment by lukasvosyka** [comment]
Hi,

the issue below is currently a blocker for us to move on to DI for functions..

https://github.com/Azure/azure-functions-host/issues/4480
